diff --git src/upstream/CMakeLists.txt src/upstream/CMakeLists.txt
index c9f4b8b..43b6394 100644
--- src/upstream/CMakeLists.txt
+++ src/upstream/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5..4.0)
 
 if (POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
@@ -539,10 +539,6 @@ if (WITH_OPENMP)
         set(WITH_SYMENGINE_THREAD_SAFE yes)
     endif()
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Clang|GNU)
-    set(CMAKE_CXX_FLAGS_DEBUG
-        "${CMAKE_CXX_FLAGS_DEBUG} -Wno-unknown-pragmas")
-    set(CMAKE_CXX_FLAGS_RELEASE
-        "${CMAKE_CXX_FLAGS_RELEASE} -Wno-unknown-pragmas")
 endif()
 
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
diff --git src/upstream/cmake/SymEngineConfig.cmake.in src/upstream/cmake/SymEngineConfig.cmake.in
index dbfc80b..8f4ef85 100644
--- src/upstream/cmake/SymEngineConfig.cmake.in
+++ src/upstream/cmake/SymEngineConfig.cmake.in
@@ -31,7 +31,7 @@
 # target_link_libraries(example ${SYMENGINE_LIBRARIES})
 #
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5..4.0)
 
 if (POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
diff --git src/upstream/cmake/UserOverride.cmake src/upstream/cmake/UserOverride.cmake
index 523ce49..3e27a3f 100644
--- src/upstream/cmake/UserOverride.cmake
+++ src/upstream/cmake/UserOverride.cmake
@@ -8,7 +8,7 @@
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     # g++
-    set(common "-Wall -Wextra -Wno-unused-parameter -fno-common")
+    set(common "-Wall -Wextra -fno-common")
     set(CMAKE_CXX_FLAGS_RELEASE_INIT "${common} -O3 -funroll-loops")
     set(CMAKE_CXX_FLAGS_DEBUG_INIT   "${common} -g -ggdb")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
@@ -18,7 +18,7 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     set(CMAKE_CXX_FLAGS_DEBUG_INIT   "${common} -g -O0")
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Clang)
     # clang
-    set(common "-Wall -Wextra -Wno-unused-parameter")
+    set(common "-Wall -Wextra")
     set(CMAKE_CXX_FLAGS_RELEASE_INIT "${common} -O3 -funroll-loops")
     set(CMAKE_CXX_FLAGS_DEBUG_INIT   "${common} -g -ggdb")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "PGI")
diff --git src/upstream/cmake/cotire.cmake src/upstream/cmake/cotire.cmake
index 741d690..e29c180 100644
--- src/upstream/cmake/cotire.cmake
+++ src/upstream/cmake/cotire.cmake
@@ -37,7 +37,7 @@ set(__COTIRE_INCLUDED TRUE)
 if (NOT CMAKE_SCRIPT_MODE_FILE)
 	cmake_policy(PUSH)
 endif()
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5..4.0)
 if (NOT CMAKE_SCRIPT_MODE_FILE)
 	cmake_policy(POP)
 endif()
diff --git src/upstream/symengine/mp_class.h src/upstream/symengine/mp_class.h
index c42d368..12d103e 100644
--- src/upstream/symengine/mp_class.h
+++ src/upstream/symengine/mp_class.h
@@ -74,12 +74,12 @@ typedef mpq_class rational_class;
 inline namespace literals
 {
 //! Literal for creating multiple precision integers
-inline integer_class operator"" _z(const char *str)
+inline integer_class operator""_z(const char *str)
 {
     return integer_class(str);
 }
 
-inline rational_class operator"" _q(const char *str)
+inline rational_class operator""_q(const char *str)
 {
     return rational_class(integer_class(str));
 }
@@ -1222,6 +1222,9 @@ inline int mp_cmpabs(const integer_class &a, const integer_class &b)
     return mpz_cmpabs(get_mpz_t(a), get_mpz_t(b));
 }
 
+// from Rmath.h
+extern "C" double unif_rand(void);
+
 class mp_randstate
 {
 public:
@@ -1241,7 +1244,8 @@ public:
     mp_randstate()
     {
         gmp_randinit_default(_state);
-        gmp_randseed_ui(_state, std::rand());
+        unsigned long int randres =  (int)(unif_rand() * (RAND_MAX - 1.0));
+        gmp_randseed_ui(_state, randres);
     }
 
     ~mp_randstate()
